hail/src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -144,6 +144,9 @@ object Simplify {
       case ArrayMap(ArrayMap(a, n1, b1), n2, b2) =>
         ArrayMap(a, n1, Let(n2, b1, b2))
 
+      case x@ArrayMap(MakeArray(args, typ), name, body) =>
+        MakeArray(args.map(a => Subst(body, Env.empty[IR].bind(name -> a))), x.typ)
```

Comment:
Whoa!  `body` might be huge and the MakeArray might  be long, this is not OK.  I'm OK with this if `args` has length 1.

Reply:
yeah, meant to flag this as something to discuss. will remove